src/pages/login/login.jsx

```diff
@@ -0,0 +1,65 @@
+import { Navbar } from "../../component/navbar/navbar"
+import { Footer } from "../../component/footer/footer"
+import axios from "axios";
+import "../login/login.css"
+import { React, useReducer } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../context/authContext";
+import { loginReducer } from "../../reducer/authReducer";
+
+export const Login = () => {
+  const navigate = useNavigate();
+  const {authDispatch}= useAuth();
+  const [{email,password},loginDispatch] = useReducer(loginReducer, {email:"", password:""});
+  const testHandler=()=>[
+    loginDispatch({type:"SET_EMAIL",payload:"rohitabhishek318@gmail.com"}),
+    loginDispatch({type:"SET_PASSWORD",payload:"rohitabhishek"})
+  ]
+  const submitHandler= async (e, email, password)=>{
+  e.preventDefault();
+  try{
+  const loginResponse = await axios.post("api/auth/login", {email,password});
+  localStorage.setItem("token", loginResponse.data.encodedToken)
+  localStorage.setItem('userData', JSON.stringify(loginResponse.data.foundUser));
+  authDispatch({ type: "USER_LOGIN" })
+  authDispatch({ type: "USER_TOKEN", payload: loginResponse.data.encodedToken })
+  authDispatch({ type: "USER_DATA", payload: loginResponse.data.foundUser })
+  }catch(err){
+    console.log(err);
+  }
+  finally{
+    navigate("/product-listing")
+  }
```

Comment:
suggestion:
You can define all API calls in a single folder.

Reply:
_Thanks will do that_